### ISSUE TITLE
Request dependency causing NSP exception

### ISSUE THREAD

**Issue by blacksun1** [twilio/twilio-node#319]
**Version:**

3.11.3

### Code Snippet

```bash
npx nsp check
```

### Exception/Log
```
┌────────────┬────────────────────────────────────────────────────────────────────┐
│            │ Prototype pollution attack                                         │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Name       │ hoek                                                               │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ CVSS       │ 4 (Medium)                                                         │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Installed  │ 4.2.0                                                              │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Vulnerable │ <= 4.2.0 || >= 5.0.0 < 5.0.3                                       │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Patched    │ > 4.2.0 < 5.0.0 || >= 5.0.3                                        │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Path       │ REMOVED > twilio@3.11.3 > request@2.83.0 > hawk@6.0.2              │
│            │ > hoek@4.2.0                                                       │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ More Info  │ https://nodesecurity.io/advisories/566                             │
└────────────┴────────────────────────────────────────────────────────────────────┘
```

### Steps to Reproduce

```bash
npx nsp check
```

### Notes

Hi,

Our build pipeline picked up the the Node Security Project issue of "Prototype pollution attack" in Hoek (https://nodesecurity.io/advisories/566) this morning. We have temporarily put in place an exception like the following

`.nsprc`

```
{
  "exceptions": [
    "https://nodesecurity.io/advisories/566"
  ]
}
```

This will require the request module to be updated first, there is already a ticket logged there (https://github.com/request/request/issues/2874)

Here is a copy of the issue:

> **Overview**
>
> The merge function, and the applyToDefaults and applyToDefaultsWithShallow functions which leverage merge behind the scenes, are vulnerable to a prototype pollution attack when provided an unvalidated payload created from a JSON string containing the __proto__ property.
> 
> This can be demonstrated like so:
> 
>     var Hoek = require('hoek');
>     var malicious_payload = '{"__proto__":{"oops":"It works !"}}';
>     
>     var a = {};
>     console.log("Before : " + a.oops);
>     Hoek.merge({}, JSON.parse(malicious_payload));
>     console.log("After : " + a.oops);
>
> This type of attack can be used to overwrite existing properties causing a potential denial of service.

Thanks in advance,

**Comment by efossier** [comment]
@blacksun1 We've created this PR (https://github.com/twilio/twilio-node/pull/324) which upgrades hoek to a patched version (4.2.1) but as a result have to drop node 0.12 as travis shows hoek 4.2.1 doesnt like node 0.12.

**Comment by efossier** [comment]
Fixed
